### ISSUE TITLE
Improve auto-import sorting rules to be better compatible with isort

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#1675]
Pylance: `v2021.8.1`

--
It would be great if `Auto-Import` could use the same default sorting rules as `isort` when adding a new import. That way the `Sort Imports` command wouldn't reorder them. In particular full support for the `order-by-type` option, i.e.
1. `CONSTANT_VARIABLE`
2. `CamelCaseClass`
3. `variable_or_function`

https://pycqa.github.io/isort/docs/configuration/options.html#order-by-type

### Current Example
```py
# lib.py

DEVICE_CLASS_BATTERY = "battery"
DEVICE_CLASS_MOTION = "motion"

def a_func(): ...
def a_func2(): ...

class BinarySensorEntity: ...
class BinarySensorEntityDescription: ...
```

```py
# sensor.py

from lib import (
    DEVICE_CLASS_BATTERY,
)

DEVICE_CLASS_BATTERY
```

```py
# using Auto-Import for the following symbols

DEVICE_CLASS_MOTION
BinarySensorEntity
BinarySensorEntityDescription
a_func
```

```py
# results in

from lib import (
    BinarySensorEntity,
    BinarySensorEntityDescription,
    DEVICE_CLASS_BATTERY,
    DEVICE_CLASS_MOTION,
    a_func,
)
```

Here `BinarySensorEntity` and `BinarySensorEntityDescription` should be put **after** `DEVICE_CLASS_MOTION`.

**Comment by jakebailey** [comment]
Are you looking for us to read an `isort` configuration, or just change our behavior to more closely match `isort`'s defaults? In general, we don't read configurations of other tools.

Note that our current behavior is similar to TypeScript's, where it inserts in a specific location, but then it's up to a formatter/linter to change it on save.

**Comment by erictraut** [comment]
I would guess that most developers use the default settings for `isort`, so I think it's reasonable that we should adopt that behavior. That was my intent when I wrote the logic that determines where pylance inserts an auto-import symbol, but I didn't realize at that time that `isort` treated all-caps symbols differently. It would be an easy fix.

**Comment by jakebailey** [comment]
If that's the default in isort, then yeah, I think we should make it match. The docs seem to imply that it's on by default.

**Comment by cdce8p** [comment]
Obviously reading the config would be great, but that is not really necessary.
I think just matching the default would be great for most devs.

**Comment by heejaechang** [comment]
it should be fixed in 2021.8.2 released today (2021.8.19)
